torch/_prims/executor.py

```diff
@@ -13,6 +13,16 @@
     from torch._C._nvfuser import Fusion, FusionDefinition  # type: ignore[import]
 
 
+def static_vars(**kwargs):
+    def decorate(func):
+        for k in kwargs:
+            setattr(func, k, kwargs[k])
+        return func
+
+    return decorate
+
+
+@static_vars(fusion_cache={})
```

Comment:
I think there isn't a need for a cache if you integrate directly with torchdynamo. It might be worth going there for the E2E

Reply:
Is there a list of functionality that torchdynamo provides?

Reply:
I'm not sure how I would give such a list; it's more about understanding the compilation model torchdynamo has. The compilation callback that dynamo calls once it has acquired an FX graph is expected to return a Python callable corresponding to the "compiled" version of the code; dynamo will take that return result and directly save it in its cache (see https://github.com/pytorch/torchdynamo/#guards ) so the next time the guards all match we will jump straight to this callable.

To get a better sense for what dynamo does, I'd recommend reading the README in the repo. You could also watch my livestream about it https://www.youtube.com/watch?v=egZB5Uxki0I if you like that sort of thing

Reply:
I think it's not right to think about this as a coverage thing, because in fact dynamo captures less stuff than `make_fx`. It is a soundness thing: you can always capture a trace with make fx but dynamo will tell you under what conditions the trace is sound to reuse later, because the Python code might execute differently and give you a different trace later. Neither dynamo nor make_fx will create a graph with numpy calls in it, but dynamo will graph break whereas make_fx will bake in a tensor constant corresponding to the computed numpy result.